### PR TITLE
Add Boost.Range adapters for Kokkos::View

### PR DESCRIPTION
**Pull request by dalg24**

### CODE REVIEW

File: packages/Search/test/tstBoostRangeAdapters.cpp [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+#include <boost/range/algorithm.hpp> // reverse_copy, replace_if, count
+#include <boost/range/numeric.hpp>   // accumulate
+
+#include "DTK_BoostRangeAdapters.hpp"
+
+#include <sstream>
+
+TEUCHOS_UNIT_TEST( BoostGeometryAdapters, Range )
+{
+    Kokkos::View<int[4], Kokkos::HostSpace> w( "w" );
```
Comment (Rombur) [inline packages/Search/test/tstBoostRangeAdapters.cpp]: Can you add a test where you work with something else than a POD?

Comment (dalg24) [reply packages/Search/test/tstBoostRangeAdapters.cpp]: I amended changes.   Computing an approximation to pi :)

File: packages/Search/test/tstBoostRangeAdapters.cpp [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    double const radius = 1.;
+    // 4/3 pi 1^3 / 2^3
+    double const pi = 6. *
+                      static_cast<double>( boost::count_if(
+                          cloud,
+                          [origin, radius]( Point point ) {
+                              return ( distance( point, origin ) <= radius );
+                          } ) ) /
+                      static_cast<double>( n );
+    std::cout << pi << "\n";
```
Comment (Rombur) [inline packages/Search/test/tstBoostRangeAdapters.cpp]: Do you really want to print this on the screen?

Comment (dalg24) [reply packages/Search/test/tstBoostRangeAdapters.cpp]: Nah I forgot to remove it :)

**Pull request merged**